tests/functional/parser/test_kafka.py

```diff
@@ -0,0 +1,88 @@
+from typing import List
+
+from aws_lambda_powertools.utilities.parser import envelopes, event_parser
+from aws_lambda_powertools.utilities.parser.models import (
+    KafkaMskEventModel,
+    KafkaRecordModel,
+    KafkaSelfManagedEventModel,
+)
+from aws_lambda_powertools.utilities.typing import LambdaContext
+from tests.functional.parser.schemas import MyLambdaKafkaBusiness
+from tests.functional.utils import load_event
+
+
+@event_parser(model=MyLambdaKafkaBusiness, envelope=envelopes.KafkaEnvelope)
+def handle_lambda_kafka_with_envelope(event: List[MyLambdaKafkaBusiness], _: LambdaContext):
+    assert event[0].key == "value"
+    assert len(event) == 1
+
+
+@event_parser(model=KafkaSelfManagedEventModel)
+def handle_kafka_event(event: KafkaSelfManagedEventModel, _: LambdaContext):
+    return event
+
+
+def test_kafka_event_with_envelope():
+    event = load_event("kafkaEventMsk.json")
+    handle_lambda_kafka_with_envelope(event, LambdaContext())
```

Comment:
We're missing a test for an envelope with `kafkaEventSelfManaged.json`. The envelope is using `KafkaMskEventModel` to parse the obj, and I'm not sure if this will work when you use the SelfManaged envelope piece.

Reply:
Thanks for pointing out this @heitorlessa! The envelope doesn't actually work with a KafkaSelfManagedEvent.

![image](https://user-images.githubusercontent.com/4295173/189083126-01479393-cd8f-4a1c-a4c2-4cf90d5f5dee.png)

Reply:
good catch!
However, i want to bring back the issue of the envelope, i think it's not reauired at all here. you'd want to use all the data, topic name etxc, not just the value payload. 
WDYT?

Reply:
Hi @ran-isenberg, I understand your concern about this, but since it's a library and widely used, we can't predict every use case.
Keeping this doesn't cause a breaking change or even make things slower than expected, so it's okay to keep that. I made a small change to support both events.